### PR TITLE
ci: modernize github actions

### PR DESCRIPTION
**Pull request by mcmxcdev**
#### Pre-flight Checklist

Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/HEAD/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/HEAD/CODE_OF_CONDUCT.md) that this project adheres to.

#### Description of Change
- use `ubuntu-22.04` runner instead of `ubuntu-20.04`
- use `windows-2022` runner instead of `windows-2019`
- use `pnpm/action-setup` instead of handling `pnpm install` manually
- use `pnpm-lock.yaml` as caching key for `actions/cache` instead of inexistent `package-lock.json`
- define `packageManager` config option in `package.json` which is read by `pnpm/action-setup`

#### Motivation and Context
Switch to the latest GitHub-hosted runners based on: https://docs.github.com/en/actions/using-jobs/choosing-the-runner-for-a-job#choosing-github-hosted-runners

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`pnpm prepare-code`)
- [x] `pnpm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

**Comment by mcmxcdev** [comment]
I am not familiar with the location of `pnpm.cjs` file on Windows, so would need help with debugging the Windows job: https://github.com/ferdium/ferdium-app/actions/runs/5668907441/job/15360556275?pr=1287

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -128,15 +129,14 @@ jobs:
...
-    name: "macos ${{ github.event.inputs.message }}"
+    name: 'macos ${{ github.event.inputs.message }}'
     needs: check_updates
     if: ${{ (needs.check_updates.outputs.should_run != 'false') && (github.event_name != 'workflow_dispatch' || (github.event_name == 'workflow_dispatch' && (contains(github.event.inputs.message, '[macOS]') || (!contains(github.event.inputs.message, '[macOS]') && !contains(github.event.inputs.message, '[Linux]') && !contains(github.event.inputs.message, '[Windows]'))))) }}
     runs-on: macos-12
     steps:
       - name: Set env vars
         run: |
           echo "NPM_CACHE=$HOME/.npm" >> $GITHUB_ENV
-          echo "PNPM_CACHE=$HOME/.pnpm-store" >> $GITHUB_ENV
```
Comment (vraravam) [inline .github/workflows/builds.yml:139]: Why are we removing the pnpm cache?

File: .github/workflows/builds.yml [hunk 2]
```diff
@@ -240,23 +240,24 @@ jobs:
...
           path: ${{ env.ELECTRON_CACHE }}
       - name: Cache electron-builder modules
         uses: actions/cache@v3
         env:
           cache-name: cache-electron-builder-modules
         with:
-          key: ${{ runner.os }}-${{ env.cache-name }}-${{ secrets.CACHE_VERSION }}-${{ hashFiles('./package-lock.json') }}
+          key: ${{ runner.os }}-${{ env.cache-name }}-${{ secrets.CACHE_VERSION }}-${{ hashFiles('./pnpm-lock.yaml') }}
           path: ${{ env.ELECTRON_BUILDER_CACHE }}
+      - name: Install pnpm
```
Comment (vraravam) [inline .github/workflows/builds.yml:252]: Should this come after the node setup?

File: .github/workflows/builds.yml [hunk 3]
```diff
@@ -305,16 +306,15 @@ jobs:
         shell: bash
 
   build_windows:
-    name: "windows ${{ github.event.inputs.message }}"
+    name: 'windows ${{ github.event.inputs.message }}'
     needs: check_updates
     if: ${{ (needs.check_updates.outputs.should_run != 'false') && (github.event_name != 'workflow_dispatch' || (github.event_name == 'workflow_dispatch' && (contains(github.event.inputs.message, '[Windows]') || (!contains(github.event.inputs.message, '[macOS]') && !contains(github.event.inputs.message, '[Linux]') && !contains(github.event.inputs.message, '[Windows]'))))) }}
-    runs-on: windows-2019
+    runs-on: windows-2022
```
Comment (vraravam) [inline .github/workflows/builds.yml:312]: Is this what's causing the windows build to fail?

File: .github/workflows/builds.yml [hunk 4]
```diff
@@ -305,16 +306,15 @@ jobs:
...
     steps:
       - name: Set env vars
         run: |
           echo "HOME=$USERPROFILE" >> $GITHUB_ENV
           echo "NPM_CACHE=$USERPROFILE/.npm" >> $GITHUB_ENV
-          echo "PNPM_CACHE=$USERPROFILE/.pnpm-store" >> $GITHUB_ENV
```
Comment (vraravam) [inline .github/workflows/builds.yml:317]: why are we deleting the pnpm cache?

File: .github/workflows/builds.yml [hunk 5]
```diff
@@ -340,27 +340,28 @@ jobs:
...
-          key: ${{ runner.os }}-${{ env.cache-name }}-${{ secrets.CACHE_VERSION }}-${{ hashFiles('./package-lock.json') }}
+          key: ${{ runner.os }}-${{ env.cache-name }}-${{ secrets.CACHE_VERSION }}-${{ hashFiles('./pnpm-lock.yaml') }}
           path: ${{ env.ELECTRON_BUILDER_CACHE }}
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2
       - name: Use Node.js specified in the '.nvmrc' file
         uses: actions/setup-node@v3
         with:
-          node-version-file: ".nvmrc"
-      - name: Install pnpm
```
Comment (vraravam) [inline .github/workflows/builds.yml:356]: If we remove these 2 lines, then will there be a possibility that we run with different version of pnpm?

**Review by vraravam — CHANGES_REQUESTED** [review]

**Review by vraravam — APPROVED** [review]

**Comment by mcmxcdev** [comment]
Cool, thanks for finishing this one up Vijay!

**Pull request merged**